source/common/version/BUILD

```diff
@@ -60,11 +60,12 @@ envoy_cc_library(
     srcs = ["version.cc"],
     copts = envoy_select_boringssl(
         [
-            "-DENVOY_SSL_VERSION=\\\"BoringSSL-FIPS\\\"",
+            "-DENVOY_SSL_VERSION=\\\"BoringSSL\\\"",
             "-DENVOY_SSL_FIPS",
         ],
         ["-DENVOY_SSL_VERSION=\\\"BoringSSL\\\""],
     ),
+    external_deps = ["ssl"],
```

Comment:
the SSL external_deps should be kept minimum usage outside of extensions, quic. Not sure pulling it here is a good idea... I would rather do an inspect in the test to avoid the misconfiguration instead of in the version library.

Reply:
Yeh I thought the dependency might be an issue. I agree that moving this into the test would be an improvement over just relying on the define set in the Bazel selection, but it's still an indirection (ie. envoy is not directly inspecting the FIPS mode), which I was trying to avoid.

Reply:
@lizan What if we move the FIPS status out of the version info and into the SSL utilities library? The server could inspect that directly to toggle the FIPS metrics. Not sure about the FIPS string in the version; maybe just rely on the define, or annotate directly in the CLI handler?

Reply:
> @lizan What if we move the FIPS status out of the version info and into the SSL utilities library? The server could inspect that directly to toggle the FIPS metrics. Not sure about the FIPS string in the version; maybe just rely on the define, or annotate directly in the CLI handler?

I didn't get what you are suggesting, rely on the define for version string is what we do today?

Reply:
I looked at more of the FIPS code yesterday and there's lots of places that deal with BoringSSL FIPS, so I'm starting to agree with that this change isn't especially useful :)

I'll close this out and update the test per your suggestion above.